### PR TITLE
chore: Resolve flaky integration tests

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Preventing some step to run through their logic if there is no data
- Fix soft delete and restore repository method to handle cases where there is no filters

### CODE REVIEW

File: packages/core/core-flows/src/product/workflows/update-products.ts [hunk 1]
```diff
@@ -39,6 +39,10 @@ function prepareUpdateProductInput({
   input: WorkflowInput
 }): UpdateProductsStepInput {
   if ("products" in input) {
+    if (!input.products.length) {
+      return {} as UpdateProductsStepInput
```
Comment (riqwan) [inline packages/core/core-flows/src/product/workflows/update-products.ts]:
```suggestion
      return {
        products: []
      }
```

**Review by riqwan — APPROVED** [review]
💪🏻

**Review by olivermrbl — APPROVED** [review]
nice!

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Review by sradevski — APPROVED** [review]

**Comment by olivermrbl** [comment]
@adrien2p, can we merge this?

**Comment by adrien2p** [comment]
@olivermrbl yes for sure

**Pull request merged**